packages/schematics/angular/guard/index.ts

```diff
@@ -35,8 +35,10 @@ export default function (options: GuardOptions): Rule {
       throw new SchematicsException('Option "implements" is required.');
     }
 
-    const implementations = options.implements.join(', ');
-    let implementationImports = `${implementations}, `;
+    const implementations = options.implements
+      .map(implement => implement === 'CanDeactivate' ? 'CanDeactivate<any>' : implement)
```

Comment:
```suggestion
      .map(implement => implement === 'CanDeactivate' ? 'CanDeactivate<unknown>' : implement)
```
`any` should be avoided where possible to prevent generation from causing build and/or lint failures.

Reply:
👍 Done!